packages/curve-ui-kit/src/features/connect-wallet/lib/ConnectionContext.tsx

```diff
@@ -118,34 +123,38 @@ export const ConnectionProvider = <
         if (walletChainId && walletChainId !== chainId) {
           setConnectState({ status: LOADING, stage: SWITCH_NETWORK })
           if (!(await setChain({ chainId: ethers.toQuantity(chainId) }))) {
-            if (abortController.signal.aborted) return
+            if (signal.aborted) return
             setConnectState({ status: FAILURE, stage: SWITCH_NETWORK })
             onChainUnavailable([chainId, walletChainId as TChainId])
           }
         }
 
         const prevLib = libRef.get<TLib>()
+        let newLib = prevLib
```

Comment:
What's the issue you were facing here that you're introducing a new local variable that `await hydrate(libRef.require<TLib>(), prevLib, wallet)` no longer worked? Was there a race condition somewhere?

Reply:
I didn't see a race happening but I think this is more correct